lib/validators/helpers.js

```diff
@@ -56,6 +56,82 @@ api.createDidDocumentLoader = ({basisBlockHeight, ledgerNode, record}) => {
   };
 };
 
+api.getValidatorParameterSet = async ({
+  basisBlockHeight, ledgerNode, validatorParameterSet
+}) => {
+  const {record} = await ledgerNode.records.get(
+    {maxBlockHeight: basisBlockHeight, recordId: validatorParameterSet});
+  return record;
+};
+
+api.validateService = async ({
+  basisBlockHeight, ledgerNode, service, validatorParameterSet
+}) => {
+  let allowedServiceBaseUrl;
+  try {
+    ({allowedServiceBaseUrl} = await api.getValidatorParameterSet({
+      basisBlockHeight, ledgerNode, validatorParameterSet
+    }));
+  } catch(error) {
+    // let NotFoundError pass
+    if(error.name !== 'NotFoundError') {
+      return {
+        error: new BedrockError(
+          'Unable to retrieve ValidatorParameterSet.', 'UnknownError', {
+            basisBlockHeight,
+            // do not expose unknown error details to the user
+            public: false,
+            httpStatusCode: 400,
+            validatorParameterSet,
+          }, error),
+        valid: false
+      };
+    }
+  }
+
+  if(!allowedServiceBaseUrl) {
+    return {
+      error: new BedrockError(
+        'Invalid ledger validator configuration. `allowedServiceBaseUrl` ' +
+        'is not defined.', 'InvalidStateError', {
+          basisBlockHeight,
+          httpStatusCode: 400,
+          public: true,
+          validatorParameterSet,
+        }),
+      valid: false
+    };
+  }
+
+  const result = api.validateServiceEndpoints({allowedServiceBaseUrl, service});
+  if(!result.valid) {
+    return result;
+  }
+  // success
+  return {valid: true};
+};
+
+api.validateServiceEndpoints = ({allowedServiceBaseUrl, service}) => {
+  let valid = false;
+  for(const {serviceEndpoint} of service) {
+    valid = allowedServiceBaseUrl.some(baseUrl =>
+      serviceEndpoint.startsWith(baseUrl));
```

Comment:
This is fine for now if we're being more lax for a reason, but we should add a FIXME indicating that it's insufficient. Once a matching `baseUrl` is found, a more rigorous check is required: ```(serviceEndpoint === `${baseUrl}/encodeURIComponent(${did})`)```. We have to enforce this format or PII can slip into the URL after the `baseUrl`.

Reply:
Fixed.